tests/test_metrics_encoder.py

```diff
@@ -15,11 +15,8 @@
 # pylint: disable=protected-access
 import unittest
 
-from snowflake.telemetry._internal.encoder.otlp.proto.common.metrics_encoder import (
-    _encode_metrics,
-)
-from snowflake.telemetry._internal.exporter.otlp.proto.metrics import (
-    ProtoMetricExporter,
+from opentelemetry.exporter.otlp.proto.common.metrics_encoder import (
+    encode_metrics,
```

Comment:
What do we test here?

Reply:
We're using the mock data helpfully created here to:

1. Make sure that the behavior of our custom 1.12.0-based _encode_metrics() method matches the behavior of the new 1.23.0 encode_metrics() method from the opentelemetry lib
2. Exercise ProtoMetricExporter._serialize_metrics_data() to do a basic sanity check on the bytes it produces